psydac/api/ast/linalg.py

```diff
@@ -85,7 +85,21 @@ def __new__(cls, ndim, block_shape, comm=None, **kwargs):
     @classmethod
     @lru_cache(maxsize=32)
     def __hashable_new__(cls, ndim, block_shape, comm_id=None, **kwargs):
+        """ Generate the Matrix Vector Product function for a BlockMatrix,StencilMatrix or StencilInterfaceMatrix.
+            In case of a BlockMatrix we give the number of blocks along the rows and columns specified with the block_shape.
+            In case of StencilMatrix or StencilInterfaceMatrix the block_shape = (1,1).
 
+        Parameters
+        ----------
+        ndim : int
+            Number of dimensions
+
+        block_shape: tuple of ints
+            The number of blocks along the rows and columns
+
+        comm_id: int
+            The communicator address
+        """
```

Comment:
Thanks for this docstring! But you should have commented the class `LinearOperatorDot` instead of this private method :stuck_out_tongue_winking_eye:

Reply:
done in [ec10ae7](https://github.com/pyccel/psydac/pull/213/commits/ec10ae70e2106c6cc339a5aa2ae5ac81c351a7f6)

Reply:
Please update docstring: the argument of the constructor is `comm`, which is an `mpi4py` (inter/intra?) communicator.

Reply:
done [0120e86](https://github.com/pyccel/psydac/pull/213/commits/0120e86010a539f112ef8f797f3edd8350d4b32f)

Reply:
So this is always an inter-communicator? Even for interfaces?

Reply:
Yes, even for the interface, the inter-communicator can be split into two intra-communicators, and I use the one that is owned by the process

Reply:
OK, then there is a typo in the docstring:

inter-communicator -> intra-communicator

Reply:
Fixed in 7426fa50b3353eb00a528005c2fb2be977f6a433